manage.py

```diff
@@ -39,55 +35,128 @@ def create_or_mask(N,ids):
     mask = mm.repmat(mask,N,1)          
     return np.logical_or(mask,mask.T)
 
-def get_node_ranks(A,rank_opts):
+def get_info_to_rank(rank_opts):
+    # returns
+    #   method to rank nodes this is a function that
+    #   takes adj mat A as an argument and returns a iterable of integer-vectors
+    #   that rank all known nodes
     
     if rank_opts['type']=='random': # select node uniformly at random
-        ranks = np.random.permutation(len(A))
+        fn = lambda A: [np.random.permutation(len(A))]
         
     if rank_opts['type']=='evec': # select node by eigenvector centrality
-        _,u = get_dom_eval_and_evec(A)
-        ranks = np.argsort(-u)
+        # id_set here passed as all nodes in A
+        # returned in a list as supposed to return iterable of iterables
+        fn = lambda A: [rank_subset_by_evec(A, id_set=np.arange(len(A)))]
             
     if rank_opts['type']=='distributed_evec': # select node by distributed eigenvector centralities
-        ranks = distributed_rank_by_evec(A,rank_opts['id_sets'])
+        info_ids = rank_opts['info_id_sets']
+        fn = lambda A: map(lambda ids: rank_subset_by_evec(A, ids), info_ids)
         
     if rank_opts['type']=='custom': # allow input of any function with any arguments
-        ranks = rank_opts['function'](A,rank_opts['arguments'])
-        
-    return ranks 
+        fn = lambda A: rank_opts['function'](A,rank_opts['arguments'])
+
+    return fn
+
+def get_info_to_control(info_to_rank, rank_opts):
+    # returns
+    #   method to return ranked control nodes this is a function that
+    #   takes adj mat A as an argument and returns a iterable of integer-vectors
+    #   that rank all known nodes
+    
+    fn = lambda A: (ir[np.isin(ir, ctl)] for ir, ctl in
+            zip(info_to_rank(A), rank_opts['control_id_sets']))
+    return fn
+
+
+def flatten_by_turns(manager_vec):
+    # return
+    #
+    # flattens over the managers in turn as would be done by
+    # numpy.matrix().flatten() if the managers vecs in columns
+    return np.array(list(chain.from_iterable(zip_longest(*manager_vec))))
```

Comment:
In theory this is a way to do it with different lengths. Need to test.

Reply:
may need to generalize this somehow:

- groups of managers controlling with a budgets
- think about this more for heterogeneous budgets